v0/destinations/wootric/util.js

```diff
@@ -0,0 +1,175 @@
+const qs = require("qs");
+const { httpGET, httpPOST } = require("../../../adapters/network");
+const {
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+const {
+  BASE_ENDPOINT,
+  VERSION,
+  ACCESS_TOKEN_CACHE_TTL_SECONDS
+} = require("./config");
+const {
+  CustomError,
+  constructPayload,
+  isDefinedAndNotNullAndNotEmpty
+} = require("../../util");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const Cache = require("../../util/cache");
+
+const ACCESS_TOKEN_CACHE = new Cache(ACCESS_TOKEN_CACHE_TTL_SECONDS);
+
+const getAccessToken = async destination => {
+  const { username, password, accountToken } = destination.Config;
+  const accessTokenKey = destination.ID;
+
+  return ACCESS_TOKEN_CACHE.get(accessTokenKey, async () => {
+    const request = {
+      header: {
+        "Content-Type": "application/x-www-form-urlencoded",
+        Accept: "application/json"
+      },
+      url: `${BASE_ENDPOINT}/oauth/token?account_token=${accountToken}`,
+      data: qs.stringify({
+        grant_type: "password",
+        username,
+        password
+      }),
+      method: "POST"
+    };
+    const wootricAuthResponse = await httpPOST(
+      request.url,
+      request.data,
+      request.header
+    );
+    const processedAuthRespone = processAxiosResponse(wootricAuthResponse);
+    // If the request fails, throwing error.
+    if (processedAuthRespone.status !== 200) {
+      throw new CustomError(
+        `[Wootric]:: access token could not be generated due to ${JSON.stringify(
+          processedAuthRespone.response
+        )}`,
+        processedAuthRespone.status
+      );
+    }
+    return processedAuthRespone.response?.access_token;
+  });
+};
+
+const retrieveUserId = async (userId, destination) => {
+  const accessToken = await getAccessToken(destination);
```

Comment:
As I see we are looking for access token multiple times? Can't we create it once and use it subsequently?

Reply:
Yes , we can . I am thinking about creating an access token in identify and track response builder then passing the same token in **util.js**.

Reply:
Updated.